specs/storage/readme.md

```diff
@@ -0,0 +1,111 @@
+# Storage module
+
+This module responsible to manage everything related with storage. In 0-OS we have 2 different storage primitives, storage pool and [0-db](https://github.com/threefoldtech/0-db).
+
+Storage pool are used when a direct disk access is required. Typical example would be a container needs to persist some data on disk.
+A storage pool is an abstract concept, it just represent something that has a path and on which we can write files.
+Usually this is implement using some kind of filesystem on top of one or multiple disks.
+
+[0-DB](https://github.com/threefoldtech/0-db) is the other storage primitives, it provide an efficient Key-Value store interface on top of a disk. 0-db implements different running mode that can server different use cases.
+
+The storage module itself is spitted into 3 sub module, each one responsible for a specific tasks.
+
+List of sub-modules:
+
+- [disks](#disk-sub-module)
+- [storage pools](#storage-pool-sub-module)
+- [0-db](#0-db-sub-module)
+
+
+## Disk sub-module
+
+Responsible to discover and prepare all the disk available on a node to be ready to use for the other sub-modules
+
+### Interface
+
+```go
+type DiskModule interface {
+    // walk over all the disks of the node
+    // create a partition table
+    // create a single partition that takes the full size of the disk
+    Prepare() (DiskInfo[],error)
```

Comment:
I think we need to take into consideration a `preparation policy` for example, in case we want to do a raid setup. In that case, a more verbose interface is needed, like what disks to use and which raid level, etc...

Reply:
Also i think, for easier storage namespace management, we can group all HDD disks in one pool, and all SSDs in another, in that case space reservation can go very fast and easy since u only have one storage pool of each type. and space checking can be very easy.

Reply:
Good idea for the preparation policy. That will give us more freedom for the future.

Regarding the grouping of the different disks, what you explain seems like an implementation detail. The interface define here doesn't really take in account disk. The only thing you need to give is a disk type and a size. How the module organize stuff underneath is not something the user needs to worry about.